### DISCUSSION TITLE
Another post about plugins

### DISCUSSION THREAD

**Discussion by BabeElena** [Pryaxis/TShock#1716] [General]
Sorry for another posts but I need some plugins can did something like this can someone recommend?

(I think some feature can did in Vanilla Tshock but I can't find it)

- Rank/Group promotion by playtime
- Cross-server chat (like who in 7777 but can chat with who in 7778)
- Item Log for hacking check (Not InvSee)
- Tshock Region Size Preview 

Thank you so much

**Comment by moisterrific** [comment]
Region size preview: region vision plugin

There are existing plugins for the other two but I don’t remember the names.

Likely all need to be updated to work on the latest TShock.

**Comment by sors89** [comment]
Rank/Group promotion by playtime: TimeRanks
Cross-server chat (like who in 7777 but can chat with who in 7778): MultiServerChat
Item Log for hacking check (Not InvSee): Hmm...
Tshock Region Size Preview: RegionVision

**Comment by Kojirremer** [comment]
Just as an addendum, you can also see a visual guide for a region's margins with vanilla TShock, provided you can see wires (either by donning equipment like The Grand Design/Mech lenses, or by holding a wiring tool). You just need to type
`/region info <NameOfTheRegion> -d`
and the region margins will be shown with wires for a few seconds.
